src/lib.rs

```diff
@@ -408,12 +416,33 @@ pub trait ConstraintSystem<E: ScalarEngine>: Sized {
 
         Namespace(self.get_root(), PhantomData)
     }
+
+    /// Most implementations of ConstraintSystem are not 'extensible': they won't implement a specialized
+    /// version of `extend` and should therefore also keep the default implementation of `is_extensible`
+    /// so callers which optionally make use of `extend` can know to avoid relying on it when unimplemented.
+    fn is_extensible() -> bool {
+        false
```

Comment:
instead of doing this, can't we have sth like 

```rust
pub trait ExtensibleConstraintSystem<E: ScalarEngine>: ConstraintSystem<E> {
  fn extend(&mut self, other: &mut Self);
}
```

Reply:
I need to dynamically depend on the result of this function downstream — so that we only take the parallel path when actually generating a groth proof. So I actually don't want to statically encode this in a trait bound. But yes, from what can be seen here, that is an alternative.